src/main/scala/org/spartanz/parserz/ParsersModule.scala

```diff
@@ -195,6 +268,27 @@ trait ParsersModule {
           from(a._2).fold(e => s -> Left(e), b => printer(value)(s, (a._1, b)))
         }
 
+      case Grammar.MapS(value, _, from) =>
+        (s: S, a: (Input, A)) => {
+          val (s1, res1) = from(s, a._2)
```

Comment:
I tend to favor using `x._n` notation sparingly.  Especially here, I think it might help with clarity (at the cost of brevity) to split `a` into `Input` and `A` values (perhaps call them `ai` and `aa` respectively).

Reply:
... In fact, I notice you doing this above.  I'd argue in favor of doing the same here.  for at the very least: consistency's sake

Reply:
@ryanonsrc I did some renaming as you suggested, thanks!